### ISSUE TITLE
perform basic validation on --email

### ISSUE THREAD

**Issue by buchdag** [zenhack/simp_le#76]
As suggested by @cpu there: https://github.com/JrCs/docker-letsencrypt-nginx-proxy-companion/issues/263#issuecomment-334252820

> simp_le should probably try to validate that the contact email being provided is valid before using it for a new-reg request

Boulder does that [with a DNS request to the MX record (with the A record as a fallback)](https://github.com/letsencrypt/boulder/blob/8aeb1a6b4d8ad4dfa4ebd2e5d55fe412349e2835/ra/ra.go#L203) of the address domain.

We can do the same or a "does it look like an email address" with regex, or combine the two.

**Comment by zenhack** [comment]
I'm leaning towards thinking checking DNS is overkill, but certainly checking the syntax makes sense.

**Comment by buchdag** [comment]
DNS in python would also require yet another package.

Should the verification soft fail by substituting something that looks incorrect by a `None` and issuing a warning ?

**Comment by zenhack** [comment]
No, If the email is syntactically invalid it should be a hard failure.

**Comment by buchdag** [comment]
Getting back to this.

Where do you think we should put the threshold for "syntactically invalid email" ?

I was thinking of, in that order:

- is it between 3 and 254 character long ? (3 is the minimum for a syntactically valid email but obviously not for an internet reachable one)
- is there an `@` in it ?
- is the local part -before the `@`- composed of one or more of any character other than these  `"(),:;<>@[\]` ? (allowed but under very specific conditions)
- is the domain part composed of one or more of `[a-z0-9.-_]` ?

Sources : RFC5322 and RFC5321 for the local part, RFC952 and RFC1123 for the domain part.

Is it already overkill ?

**Comment by zenhack** [comment]
Why not just the regex:

    .+@[a-z0-9._-]+

..which captures everything but the second-to-last bullet?

Note that I switched the order of `-` and `_` in the trailing regex,
since `-` denotes a range if it's in the middle somewhere.

I think that's probably close enough.

**Comment by zenhack** [comment]
Quoting Ian Denhardt (2017-12-03 20:04:53)

> Why not just the regex:
>
>     .+@[a-z0-9._-]+

Note that we should perhaps allow upercase letters as well; DNS is
supposed to be case-insensitive. So:

    .+@[A-Za-z0-9._-]+

**Comment by buchdag** [comment]
> I think that's probably close enough.

That suits me too.

> Note that we should perhaps allow upercase letters as well; DNS is supposed to be case-insensitive.

Yep, I forgot to mention lowercasing the email argument before the test.

So, something like this:

```
def validate_mail(email):
    
    email=email.lower()
    regex='stuff'
    if not regex.match(email):
        raise Error('bla')
    else
        return email
```

And use that in the argument parser, or just a straight `if somereg.match(args.email):` in `main_with_exceptions()` ?

**Comment by zenhack** [comment]
I'd just add the check to `main_with_exceptions`.

**Comment by buchdag** [comment]
Closed by #93

**Comment by zenhack** [comment]
Yup, thanks
